api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -446,20 +448,45 @@ private DbUser updateDuccAgreement(
   }
 
   @Override
-  public void validateTermsOfService(Integer tosVersion) {
+  public void validateAllOfUsTermsOfService(Integer tosVersion) {
```

Comment:
when this was originally written, we only checked AOU TOS when creating the user.  If the TOS was wrong, it made sense to throw a Bad Request.

But now we should be more thoughtful.  If an existing user has an outdated TOS, that's not really what I would call a Bad Request.  So I'd prefer the previous signature with Boolean

Reply:
I prefer BadRequest because it had message that helped identify why exactly the validation failed, is it a bad user who has not selected TOS (AoU or terra) in the first place  or just  outdated version . 

That could also help in showing the user a more appropriate message if we need to change the UX. 

I can make it boolean right now , so that we are unblocked and can discuss this further tomorrow